### PR TITLE
Add github workflow for publishing npm package

### PR DESCRIPTION
**Pull request by michalsmiarowski**
Adds github worklow for publishing npm package to the npm registry.

### CODE REVIEW

File: .github/workflows/npm.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    paths:
+      - ".storybook/**"
+      - "src/**"
+      - "rollup.config.ts"
+      - "package.json"
+      - "yarn.lock"
+  workflow_dispatch:
+
+jobs:
+  npm-compile-publish-contracts:
```
Comment (michalinacienciala) [inline .github/workflows/npm.yml:16]:
```suggestion
  npm-compile-publish:
```
We don't deal with contracts here, we can drop that part of the job name.

**Review by michalinacienciala — COMMENTED** [review]
Left couple of comments.
Also:
1. Do we have any package published already? I don't see anything here: https://www.npmjs.com/package/@threshold-network/components.
2. Let's bump up the version in `package.json` to `1.0.0-dev` (or, if the `1.0.0` is already published, let's change the version to higher one.
3. In `package.json` I can see that only the files from `src/` and `lib/` folders will be included in the package. That's probably right, but I just want to make sure we don't need anything else, like files from `.storybook/`.
4. It would be good to test the workflow before merging it to main. We usually do that by creating a commit that temporarily adds the `pull-request:` to the workflow triggers and adds the `--dry-run` flag to the publish step. Such commit will trigger the execution of workflow, without really publishing the package. Once we see that everything looks valid, we can remove those temporary changes.

File: .github/workflows/npm.yml (hunk 1 above)
Comment (michalsmiarowski) [reply .github/workflows/npm.yml]: 9c7395f5e0c93e8bb91d010d3c63916bafee0e3a

**Comment by michalsmiarowski** [comment]
Thanks @michalinacienciala!

1. We don't have one yet. This will be first time publishing it. Are there additional steps that should be done in that case?
2. 30efb4757dce64185dd7385cbe4933f87e87b1b7
3. Tbh I think the storybook shouldn't be included in the package (and also stories in `src` directory, so I should probably somehow exclude them). Let me do a little bit of a research though, because i'm not sure about it.
4. Got it! Will do that soon!

**Comment by michalinacienciala** [comment]
> 1. We don't have one yet. This will be first time publishing it. Are there additional steps that should be done in that case?
OK, so there are couple of things that need to be done:
- We need to reserve/activate the `@threshold-network/components` registry (I don't have the full knowledge of what exactly needs to be done here, I don't have the right privileges - Kuba, Piotr and possibly Łukasz as well should be able to do that)
- In GH settings we need to configure `NPM_TOKEN` secret to be accessible by the `threshold-network/components` (again, Kuba/Piotr/Łukasz help will be needed)
- First package (`1.0.0-dev.0`) needs to be published either manually from the command line (which again requires authorization to the NPM registry) or can be done by manipulating workflow code: once we have the npm registry and `NPM_TOKEN` configured, we can comment out the step with the version bump up and make a temporary change to the `package.json`, setting the version to (`1.0.0-dev.0`). Once we commit the change, workflow will run and publish first package. Then we can create a commit that removes that temporary change (or it would be even better to remove the commit from the git history, to not pollute the history with insignificant commits)

@nkuba, @pdyraga or @lukasz-zimnoch , could one of you help with configuring `@threshold-network/components` NPM registry and `NPM_TOKEN`?

**Review by michalinacienciala — APPROVED** [review]
Looking good, first package was successfully published!

**Comment by michalinacienciala** [comment]
I'm going to merge the PR despite the failing formatting check (we're not merging to `main` yet). Let's fix formatting in #2, since that PR is causing the problem.

**Pull request merged**